icu4c/source/i18n/units_converter.cpp

```diff
@@ -293,13 +300,16 @@ void loadConversionRate(ConversionRate &conversionRate, const MeasureUnitImpl &s
     conversionRate.factorNum = finalFactor.factorNum;
     conversionRate.factorDen = finalFactor.factorDen;
 
+    // This code corresponds to ICU4J's ConversionRates.getOffset().
     // In case of simple units (such as: celsius or fahrenheit), offsets are considered.
     if (checkSimpleUnit(source, status) && checkSimpleUnit(target, status)) {
         conversionRate.sourceOffset =
             sourceToBase.offset * sourceToBase.factorDen / sourceToBase.factorNum;
         conversionRate.targetOffset =
             targetToBase.offset * targetToBase.factorDen / targetToBase.factorNum;
     }
+    // TODO: should we consider failure if there's an offset for a
+    // not-simple-unit? What about kilokelvin / kilocelsius?
```

Comment:
kilokelvin is fine; it could be used to describe temperatures at the surface of the sun, for example.  I don't think I've ever seen a kilocelsius or even have a guess about what that could mean.

We should, at least, have a test case that demonstrates what our behavior is in edge cases like this.  Follow-up issue?

Reply:
I added a ticket for tracking handling of "undefined behaviour" - https://github.com/icu-units/icu/issues/127 - additional items can be added to it in the future.